### ISSUE TITLE
Galaxy E7 Duos 3G SM-E700H : On this device the soft keyboard and emoji pop are inflated together.

### ISSUE THREAD

**Issue by DalveerSinghDaiya** [vanniktech/Emoji#191]
Thank you for filing an issue. If this is a bug that you want to report, please take the time to provide some information:

- Version of the library:
- Affected devices:
- Affected versions:

Screenshot(s) showing the issue and reprodution steps are appreciated.

----------------------------------------------------------------------

**Comment by vanniktech** [comment]
Hmm do other applications that do this kind of thing work? E.g. WhatsApp or telegram?

**Comment by DalveerSinghDaiya** [comment]
Nopes!! Its working fine in WhatsApp and Telegram.

**Comment by vanniktech** [comment]
Hmmm :( that's not any good. I don't have the device at hand and in all the devices that I tested the library against (20+) devices. Are you willing to investigate this issue and maybe come up with a PR?

**Comment by DalveerSinghDaiya** [comment]
I have tried some tricks but all for no good, its not working, that's why i filed it here, thought some one would have cracked this already.

**Comment by vanniktech** [comment]
Yeah as you probably know detecting whether or not the keyboard shown is non trivial especially when certain devices / manufacters do their own thing. Maybe someone will crack it.

**Comment by DalveerSinghDaiya** [comment]
Haha, I was expecting you to do it :P

**Comment by vanniktech** [comment]
Nope I'm also not actively looking into it. Feel free to take this one

**Comment by vanniktech** [comment]
Related to #224.

@DalveerSinghDaiya could you try the possible workarounds that were posted in that issue and see if that would resolve your issue?

**Comment by daniele-athome** [comment]
Despite I would really prefer to keep the current behavior, I would like to point out that Hangouts (meaning it's Google doing this) closes the keyboard before showing the emoji panel. It seems like they gave up eheh :-)
If you need help to test or code something to fix this, I'm here.

**Comment by vanniktech** [comment]
We need to show the keyboard in order to get the height.

**Comment by daniele-athome** [comment]
> We need to show the keyboard in order to get the height.

Sure, I mean get the height and then close it. I believe that's what Hangouts does.

**Comment by vanniktech** [comment]
That sounds good. Happy to review an implementation for that.

**Comment by daniele-athome** [comment]
Now that I think more about it, it will become a simple EmojiView use case, meaning no EmojiPopup involved. By exposing EmojiView directly (and by implementing some helper code to use it), an app can add an EmojiView to its layout and hide the keyboard before showing it, using the appropriate height (the helper code could be used to do just that, leveraging the already implemented code in EmojiPopup used to calculate the height).
What do you say about exposing EmojiView to the world? Any other ideas or questions about this approach?

**Comment by danielgaribaldi** [comment]
Hi guys!

I'm having the same problem, both keyboards are shown at the same time on a Pixel 2XL with Android 8.1 using emoji-google:0.6.0-SNAPSHOT

![issuw](https://user-images.githubusercontent.com/2779345/36385953-7f4b449e-1594-11e8-8873-24b391341e71.png)

Did you find a solution/workaround for this?

Thanks!


ps: beside this small problem the library is awesome! thanks!

**Comment by daniele-athome** [comment]
> Did you find a solution/workaround for this?

Unfortunately not. I'm working on using EmojiView directly inside my layout and hide the virtual keyboard before showing the emoji panel.

**Comment by 186538** [comment]
the problem appears when another app uses the top level screen, try this ..
_ install esfile (https://play.google.com/store/apps/details?id=com.estrongs.android.pop)
_ activate "ES Swipe" feature

then you will get the emoji appears above the keyboard !!
this problem happens all the time with android P, i think it should be fixed as soon as possible

**Comment by danielgaribaldi** [comment]
My only solution was to fork the project and use a different approach on
how the emoji keyboard appears.

Now I use it as a fragment embedded in the activity or fragment and I play
with the height of this fragment and the normal keyboard visibility to make
it appear or disappear. Sounds a bit weird but works really well and in all
android versions in including Oreo and P

**Comment by chihung93** [comment]
@danielgaribaldi thanks for the solution.
I've created a customized view for fixing this bug.
If anyone needs my code, just check out my GitHub or look out this class.

Github:
https://github.com/vanniktech/Emoji

Customize view 
https://github.com/chihung93/Emoji/blob/master/emoji/src/main/java/com/vanniktech/emoji/EmojiUniversal.java 

P.s: Thanks @vanniktech @danielgaribaldi.

**Comment by vanniktech** [comment]
> I've created a customized view for fixing this bug.

Instead of?

**Comment by chihung93** [comment]
@vanniktech : instead of EmojiPopup

**Comment by vanniktech** [comment]
And what do you use for displaying that custom Layout?

**Comment by chihung93** [comment]
@vanniktech : Yes.  I just custom layout for fixing the problem with the popup.

**Comment by srivishnu888** [comment]
What is the solution for this?
